src/core/server/core_usage_data/types.ts

```diff
@@ -0,0 +1,154 @@
+  savedObjects: {
+    indices: Array<{
+      name: string;
+      docsCount: number;
+      docsDeleted: number;
+      storeSizeBytes: number;
+      primaryStoreSizeBytes: number;
+    }>;
+  };
+}
+
+/**
+ * Usage data on this Kibana node's runtime environment.
+ */
+export interface CoreEnvironmentUsageData {
+  memory: {
+    heapTotalBytes: number;
+    heapUsedBytes: number;
+    /** V8 heap size limit */
+    heapSizeLimit: number;
+  };
+  os: {
+    platform: string;
+    platformRelease: string;
+    distro?: string;
+    distroRelease?: string;
+  };
+}
+
+/**
+ * Usage data on this cluster's configuration of Core features
+ */
+export interface CoreConfigUsageData {
+  elasticsearch: {
+    sniffOnStart: boolean;
+    sniffIntervalMs?: number;
+    sniffOnConnectionFault: boolean;
+    numberOfHostsConfigured: number;
+    requestHeadersWhitelistConfigured: boolean;
+    customHeadersConfigured: boolean;
+    shardTimeoutMs: number;
+    requestTimeoutMs: number;
+    pingTimeoutMs: number;
+    logQueries: boolean;
+    ssl: {
+      verificationMode: 'none' | 'certificate' | 'full';
+      certificateAuthoritiesConfigured: boolean;
+      certificateConfigured: boolean;
+      keyConfigured: boolean;
+      keystoreConfigured: boolean;
+      truststoreConfigured: boolean;
+      alwaysPresentCertificate: boolean;
+    };
+    apiVersion: string;
+    healthCheckDelayMs: number;
+  };
+
+  http: {
+    basePathConfigured: boolean;
+    maxPayloadInBytes: number;
+    rewriteBasePath: boolean;
+    keepaliveTimeout: number;
+    socketTimeout: number;
+    compression: {
+      enabled: boolean;
+      referrerWhitelistConfigured: boolean;
+    };
+    xsrf: {
+      disableProtection: boolean;
+      whitelistConfigured: boolean;
+    };
+    requestId: {
+      allowFromAnyIp: boolean;
+      ipAllowlistConfigured: boolean;
+    };
+    ssl: {
+      certificateAuthoritiesConfigured: boolean;
+      certificateConfigured: boolean;
+      cipherSuites: string[];
+      keyConfigured: boolean;
+      keystoreConfigured: boolean;
+      truststoreConfigured: boolean;
+      redirectHttpFromPortConfigured: boolean;
+      supportedProtocols: string[];
+      clientAuthentication: 'none' | 'optional' | 'required';
+    };
+  };
+
+  logging: {
+    appendersTypesUsed: string[];
+    loggersConfiguredCount: number;
+  };
+
+  // plugins: {
+  //   /** list of built-in plugins that are disabled */
+  //   firstPartyDisabled: string[];
+  //   /** list of third-party plugins that are installed and enabled */
+  //   thirdParty: string[];
+  // };
+
+  savedObjects: {
+    maxImportPayloadBytes: number;
+    maxImportExportSizeBytes: number;
+  };
+
+  // uiSettings: {
+  //   overridesCount: number;
+  // };
+}
+
+/**
+ * Internal API for getting Core's usage data payload
+ * @internal
+ */
+export interface CoreUsageDataStart {
+  /**
+   * Internal API for collecting Core usage data
+   * @internal
+   * */
```

Comment:
Can we add a note like: "This API should never be used to drive application logic and is only intended for telemetry purposes"